icu4c/source/tools/genrb/reslist.cpp

```diff
@@ -1442,7 +1479,7 @@ SRBRoot::compactKeys(UErrorCode &errorCode) {
                         keys[newpos++] = keys[oldpos++];
                     }
                 }
-                assert(i == keysCount);
+                U_ASSERT(i == keysCount);
```

Comment:
FYI: In a tool, I assume that it's ok to use assert() rather than U_ASSERT(). But this is fine, too.